### ISSUE TITLE
Code won't run if external config file not present, due to conflicting defaults

### ISSUE THREAD

**Issue by jbrucker** [sirateek/ku-polls#13]
Great that you used `environ` to externalize sensitive data,
but there's a problem if the external configuration file is not found.

In `settings.py` you have:
```python
DEBUG = env('DJANGO_DEBUG', default=False)
ALLOWED_HOSTS = [ ]
```
if the configuration file is not found then DEBUG = False (this is correct)
but when DEBUG=False, Django strictly enforces ALLOWED_HOSTS
and so no one can connect to the server.

You could externalize ALLOWED_HOSTS, too,
or simply use this:
```python
ALLOWED_HOSTS = ['localhost', '127.0.0.1']
```

By the way, to make it **easy** on the user to configure things,
I suggest you use the same variable names in the configuration file
as in the Django settings.py. 

For example, externalize "DEBUG" as a variable named "DEBUG".
this is what is usually done.

```python
DEBUG = env('DEBUG', default=False)
```

If you have a complex project with **lots** of externalized settings,
then it is conventional to use "." notation and lowercase names.
This is done to avoid "name collisions".

For example:
```python
DEBUG = env("django.debug", default=False)
```
and in .env:
```
django.debug = False
```
JUnit, Log4J, and the Java compiler use this naming convention for 
configuration settings.

Good job on this assignment. --jim

**Comment by sirateek** [comment]
Thank you for pointing this issue out to me 😄
I'll make sure that this is fixed before the `iteration2` is merged. 👍

**Comment by sirateek** [comment]
It seems that the `django-environ` package ver. 0.4.5 doesn't support reading the dot notation env.
I also tested on the version 0.7.0 but the result is the same.

I looked into the [source code][django-envron-source] on version 0.7.0 and found that the pattern checker doesn't consider the dot nation pattern as an env. 

_Could this be a bug on their package?_ 🤔

<details>
  <summary>Example code</summary>

```python
import re
line = "django.debug=true"
m1 = re.match(r'\A(?:export )?([A-Za-z_0-9]+)=(.*)\Z', line)

if m1:
    key, val = m1.group(1), m1.group(2)
    m2 = re.match(r"\A'(.*)'\Z", val)
    if m2:
        val = m2.group(1)
    m3 = re.match(r'\A"(.*)"\Z', val)
    if m3:
        val = re.sub(r'\\(.)', r'\1', m3.group(1))
    print(key, val)
else:
    print("Pattern not matched")

# OUTPUT:
# Pattern not matched
```
</details>

Anyway. I'll instead use the upper case style for the env variable naming.

[django-envron-source]: https://github.com/joke2k/django-environ/blob/caf1ce28b235330fd14a8f6e6f95293e7c187247/environ/environ.py#L772~L782

**Comment by jbrucker** [comment]
When you close an issue that someone else opened, you should write a
reason why you closed it.
This was mentioned in class.

**Comment by sirateek** [comment]
> When you close an issue that someone else opened, you should write a reason why you closed it. This was mentioned in class.
> […](#)
> On Fri, Oct 8, 2021 at 3:11 PM Siratee K. ***@***.***> wrote: Closed #13 via #14. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

I'm really sorry about this. This issue was fixed and was automatically closed via the merge of PR #14 as mentioned. So I thought it was straightforward since the timeline is already showing the detail about that. Also, I already wrote the PR detail and I did mention how this issue was fixed in #14.
Anyway, Thank you again for opening this issue :)
